packages/service-provider-core/src/connection-string.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { expect } from 'chai';
+import { ConnectionString } from './connection-string';
+
+describe('ConnectionString', () => {
+  context('with valid URIs', () => {
+    for (const { uri, match } of [
+      {
+        uri: 'mongodb://localhost/',
+        match: {
+          href: 'mongodb://localhost/',
+          protocol: 'mongodb:',
+          username: '',
+          password: '',
+          pathname: '/',
+          search: '',
+          hash: '',
+          isSRV: false,
+          hosts: ['localhost']
+        }
+      },
+      {
+        uri: 'mongodb+srv://localhost',
+        match: {
+          href: 'mongodb+srv://localhost/',
+          protocol: 'mongodb+srv:',
+          username: '',
+          password: '',
+          pathname: '/',
+          search: '',
+          hash: '',
+          isSRV: true,
+          hosts: ['localhost']
+        }
+      },
+      {
+        uri: 'mongodb+srv://cat:meow@localhost/',
+        match: {
+          href: 'mongodb+srv://cat:meow@localhost/',
+          protocol: 'mongodb+srv:',
+          username: 'cat',
+          password: 'meow',
+          pathname: '/',
+          search: '',
+          hash: '',
+          isSRV: true,
+          hosts: ['localhost']
+        }
+      },
+      {
+        uri: 'mongodb://cat:meow@localhost:12345/db',
+        match: {
+          href: 'mongodb://cat:meow@localhost:12345/db',
+          protocol: 'mongodb:',
+          username: 'cat',
+          password: 'meow',
+          pathname: '/db',
+          search: '',
+          hash: '',
+          isSRV: false,
+          hosts: ['localhost:12345']
+        }
+      }
```

Comment:
Could we add one more test case with multiple hosts and a query string?
```
{
  uri: 'mongodb://localhost:12345,anotherHost/?directConnection=true',
  match: {
    href: 'mongodb://localhost:12345,anotherHost/?directConnection=true',
    protocol: 'mongodb:',
    username: '',
    password: '',
    pathname: '/',
    search: '?directConnection=true',
    hash: '',
    isSRV: false,
    hosts: ['localhost:12345', 'anotherHost']
  }
}
```

Reply:
Sure, done!